python/ray/dag/compiled_dag_node.py

```diff
@@ -357,12 +362,9 @@ def __init__(
             self.output_channels, self.output_idxs
         )
         # Store the intermediate result of a READ or COMPUTE operation.
-        # The dict is from the operation to its input, i.e., the intermediate
-        # result from the last operation. There are only two cases:
-        # if the operation is COMPUTE, the value is the result from the READ with
-        # the same exec_task_idx; if the operation is WRITE, the value is the result
-        # from the COMPUTE with the same exec_task_idx.
-        self._intermediate_buffer: Dict[_DAGNodeOperation, Any] = {}
+        # The result of a READ operation will be used by a COMPUTE operation,
+        # and the result of a COMPUTE operation will be used by a WRITE operation.
+        self._intermediate_buffer: Any = None
```

Comment:
```suggestion
        self._intermediate_buffer: DAGOperationFuture = None
```

Reply:
And update comment to say caller must call wait before using the value.

Reply:
Done.